docs/design/coreclr/botr/logging.md

```diff
@@ -0,0 +1,70 @@
+
+To emit EventPipe events from C# you can use the [`System.Diagnostics.Tracing.EventSource`](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/eventsource-getting-started) class, by defining your own event types derived from it. See its documentation for examples and more details.
+
+StressLog
+------------
+The StressLog is a circular buffer inside the runtime process that usually does not escape, and most StressLog messages are enabled in retail builds, which makes it useful for troubleshooting issues with GC or other subsystems in production scenarios. To enable it, you can set the `DOTNET_StressLog` environment variable to `1`, and you can configure it by setting the `DOTNET_LogFacility`, `DOTNET_LogFacility2` and `DOTNET_LogLevel` environment variables. If you are in a situation where you can't attach a debugger to the process, the `DOTNET_StressLogFilename` environment variable can be used to write the StressLog's contents to a file instead.
+
+To write your own messages to the StressLog from C++, you can use the `STRESS_LOG_N(facility, level, msg, ...)` macros, i.e. `STRESS_LOG1(LF_GC, LL_ERROR, "A significant but non-fatal error occurred in the garbage collector! Here's my favorite number: %d\n", 42)` where the first argument is one or more logging facilities (you can combine them using `|` i.e. `LF_GC | LF_GCROOTS`), the second argument is a severity level, and the third argument is the log message format string.
+
+You can't write your own messages to the StressLog from C#. If for some reason you specifically need to while testing, perhaps you could expose it via a custom icall. 😅
+
+Traditional .NET Runtime Logging
+--------------------------------
+"Traditional" log messages are only enabled in debug or checked builds of the runtime. To enable them, you can set the `DOTNET_LogEnable` environment variabvle to `1`, and configure them using the `DOTNET_LogFacility`, `DOTNET_LogFacility2` and `DOTNET_LogLevel` environment variables. You can set `DOTNET_LogToFile` to write the log to a file on disk instead of its default target, and you can explicitly write it to the debugger or the console using `DOTNET_LogToDebugger` or `DOTNET_LogToConsole`. In multi-process logging scenarios, you can set `DOTNET_LogWithPid` to append a process ID to all the log messages so it is easier to understand how processes are interacting.
+
+This classical logging system is not frequently used, so individual log messages you encounter may be partially or completely nonfunctional - for example, 64-bit-only issues in outdated log statements - but the basics should always work.
+
+To send your own traditional log messages from C++, you can use the `LOG((facility, level, msg, ...))` macro, i.e. `LOG((LF_GC, LL_INFO10000, "An insignificant thing happened in the garbage collector.\n"))`. The arguments are roughly equivalent to those described above for StressLog.
```

Comment:
Note that `STRESS_LOG` include `LOG` and that arguments for the `STRESS_LOG` and `LOG` macros are very similar. It tends to be easy to flip LOG to STRESS_LOG for debugging purposes, etc.